mempool/clist_mempool.go

```diff
@@ -542,23 +551,40 @@ func (mem *CListMempool) resCbFirstTime(
 func (mem *CListMempool) resCbRecheck(req *abci.Request, res *abci.Response) {
 	switch r := res.Value.(type) {
 	case *abci.Response_CheckTx:
+		tx := req.GetCheckTx().Tx
+		txHash := TxKey(tx)
+		e, ok := mem.txsMap.Load(txHash)
+
 		if r.CheckTx.Code == abci.CodeTypeOK {
-			// Good, nothing to do.
+			if !ok {
+				panic(fmt.Sprintf("Unexpected tx response from proxy during recheck\ntxHash=%s, tx=%X", txHash, tx))
```

Comment:
Depending on the timing, there are cases where it has already been deleted. So would you do logging rather than panic?